File: matlab/tagtools/tagiofuns/get_start_time.m
function       t = get_start_time(info)

%       t = get_start_time(info)

if nargin<1,
   help get_start_time
   return
end

if ~isstruct(info),
   fprintf('Argument to get_start_time must be an info structure\n') ;
   return
end

t = [] ;
if isfield(info,'dephist_device_datetime_start'),
   t = info.dephist_device_datetime_start ;
end

if isempty(t) && isfield(info,'dephist_device_datetime_start'),
   t = info.dephist_device_datetime_start ;
end

if isempty(t),
   fprintf('No valid start time in the info structure\n') ;
   return
end

if isfield(info,'dephist_device_regset'),
   r = info.dephist_device_regset ;
else
   r = 'dd/mm/yyyy HH:MM:SS' ;
end

try
   t=datevec(t,r) ;
catch
   fprintf('Unable to convert time string in info structure\n') ;
end

File: Python/tagiofuns/get_start_time.py
def get_start_time(info=None):
    
    """
    t = get_start_time(info)
    
    """
    from datetime import datetime
    import dateutil.parser
    
    t = []
    if not info:
        print(help(get_start_time))
        return t
    
    if not isinstance(info, dict):
        print('Argument to get_start_time must be an info dictionary.\n')
        return t
    
    if 'dephist_device_datetime_start' in info.keys():
        t = info['dephist_device_datetime_start']
        
    if not t and 'dephist_device_datetime_start' in info.keys():
        t = info['dephist_device_datetime_start']
        
    if not t:
        print('No valid start time in the info dictionary.\n')
        return t
    
    try:
        t = dateutil.parser.parse(t)
        ms = t.microsecond * 1e-6
        t = list(t.timetuple()[:6])
        t[-1] = t[-1] + ms
    except Exception:
        print('Unable to convert time string to a date vector/list.\n') 

    return t    